framework/src/main/java/org/checkerframework/common/aliasing/AliasingVisitor.java

```diff
@@ -271,7 +265,37 @@ private boolean canBeLeaked(Tree exp) {
         AnnotatedTypeMirror type = atypeFactory.getAnnotatedType(exp);
         boolean isMethodInvocation = exp.getKind() == Kind.METHOD_INVOCATION;
         boolean isNewClass = exp.getKind() == Kind.NEW_CLASS;
-        return type.hasExplicitAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        boolean isUniqueClassFlag = isUniqueClass(exp);
+        if (isUniqueClassFlag) {
+            return type.hasAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        } else {
+            return type.hasExplicitAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        }
+    }
+
+    /**
+     * Returns true if class of tree expression {@code exp} has type {@code @Unique}
+     *
+     * @param exp the Tree to check
+     * @return boolean true if class if unique and false otherwise
+     */
+    private boolean isUniqueClass(Tree exp) {
+        AnnotatedTypeMirror type = atypeFactory.getAnnotatedType(exp);
+        Element el = types.asElement(type.getUnderlyingType());
+        if (el != null) {
+            Set<AnnotationMirror> annoMirrors = atypeFactory.getDeclAnnotations(el);
+            if (annoMirrors != null) {
+                for (AnnotationMirror mirror : annoMirrors) {
+                    if (mirror.getAnnotationType()
+                            .asElement()
+                            .getSimpleName()
+                            .contentEquals("Unique")) {
```

Comment:
Again, don't use Strings for things like this. Can you use `AnnotationUtils.containsSameByClass` https://github.com/typetools/checker-framework/blob/master/javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java#L230 ? Then you don't need the `for` loop.

Reply:
I have used the AnnotationUtils.containtsSameByClass method in my latest commit